lib/result_set_presenter.rb

```diff
@@ -108,55 +108,9 @@ def spelling_suggestions
     @context[:spelling_suggestions]
   end
 
-  def document_series_by_slug(slug)
-    document_series = document_series_registry && document_series_registry[slug]
-    if document_series
-      document_series.to_hash.merge("slug" => slug)
-    else
-      {"slug" => slug}
-    end
-  end
-
-  def document_collection_by_slug(slug)
-    document_collection = document_collection_registry && document_collection_registry[slug]
-    if document_collection
-      document_collection.to_hash.merge("slug" => slug)
-    else
-      {"slug" => slug}
-    end
-  end
-
-  def organisation_by_slug(slug)
-    organisation = organisation_registry && organisation_registry[slug]
-    if organisation
-      organisation.to_hash.merge("slug" => slug)
-    else
-      {"slug" => slug}
-    end
-  end
-
-  def topic_by_slug(slug)
-    topic = topic_registry && topic_registry[slug]
-    if topic
-      topic.to_hash.merge("slug" => slug)
-    else
-      {"slug" => slug}
-    end
-  end
-
-  def world_location_by_slug(slug)
-    world_location = world_location_registry && world_location_registry[slug]
-    if world_location
-      world_location.to_hash.merge("slug" => slug)
-    else
-      {"slug" => slug}
-    end
-  end
-
-  def sector_by_slug(slug)
-    sector = specialist_sector_registry && specialist_sector_registry[slug]
-    if sector
-      sector
+  def structure_by_slug(structure, slug)
+    if items = structure && structure[slug]
```

Comment:
In this case I think `items` is always actually a single item.  Might be better named in the singular.